pgrx/src/pg_catalog.rs

```diff
@@ -0,0 +1,866 @@
+//LICENSE Portions Copyright 2019-2021 ZomboDB, LLC.
+//LICENSE
+//LICENSE Portions Copyright 2021-2023 Technology Concepts & Design, Inc.
+//LICENSE
+//LICENSE Portions Copyright 2023-2023 PgCentral Foundation, Inc. <contact@pgcentral.org>
+//LICENSE
+//LICENSE All rights reserved.
+//LICENSE
+//LICENSE Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+use crate as pgrx;
+use pgrx::datum::Array;
+use pgrx::pg_sys::Oid;
+use pgrx::pg_sys::Oid as Regproc;
+use std::ffi::{c_char, CStr};
+
+unsafe trait GetStruct<T> {
+    unsafe fn get_struct(raw: *const T) -> Self;
+}
+
+unsafe impl<T> GetStruct<T> for T {
+    unsafe fn get_struct(raw: *const T) -> Self {
+        unsafe { raw.read() }
+    }
+}
+
+unsafe impl GetStruct<pgrx::pg_sys::nameData> for &CStr {
+    unsafe fn get_struct(raw: *const pgrx::pg_sys::nameData) -> Self {
+        unsafe { CStr::from_ptr(raw.cast::<c_char>()) }
+    }
+}
+
+unsafe impl GetStruct<pgrx::pg_sys::int2vector> for &[i16] {
+    unsafe fn get_struct(raw: *const pgrx::pg_sys::int2vector) -> Self {
+        unsafe { (*raw).values.as_slice((*raw).dim1 as usize) }
+    }
+}
+
+unsafe impl GetStruct<pgrx::pg_sys::oidvector> for &[Oid] {
+    unsafe fn get_struct(raw: *const pgrx::pg_sys::oidvector) -> Self {
+        unsafe { (*raw).values.as_slice((*raw).dim1 as usize) }
+    }
+}
+
+macro_rules! _macro_1 {
```

Comment:
Okay, these macros definitely need names, not ordinals.

Reply:
Renamed to `define_character`.

Reply:
Thank you!